src/main/java/com/microsoft/aad/adal4j/AdalOAuthRequest.java

```diff
@@ -108,7 +108,17 @@ void configureHeaderAndExecuteOAuthCall(final HttpURLConnection conn)
             conn.setRequestProperty("Authorization", this.getAuthorization());
         }
 
-        HttpHelper.configureAdditionalHeaders(conn, this.extraHeaderParams);
+        Map<String, String> params = new java.util.HashMap<>();
+        if (this.extraHeaderParams != null && !this.extraHeaderParams.isEmpty()) {
+            for (java.util.Map.Entry<String, String> entry : this.extraHeaderParams.entrySet()) {
+                if (entry.getValue() == null || entry.getValue().isEmpty()) {
+                    continue;
+                }
+                params.put(entry.getKey(), entry.getValue());
+            }
+        }
+        //HttpHelper.configureAdditionalHeaders(conn, this.extraHeaderParams);	// Fixed Google Appengine URLFetch Exception
+        HttpHelper.configureAdditionalHeaders(conn, params);

```

Comment:
You should make this fix in HttpHelper.configureAdditionalHeaders with the following change:

```java
if(!StringHelper.isBlank(entry.getValue())){
          params.put(entry.getKey(), entry.getValue());
 }
```